AppCenter/AppCenterSDK+HiDrive.swift

```diff
@@ -0,0 +1,150 @@
+
+
+fileprivate enum AppCenterConstants {
+
+	static let appSecretKey					= "AppCenterAppSecret"
+	static let smfLogUploadMaxSize	: Int	= 5000
+}
+
+class AppCenterSDK: NSObject {
+
+	// MARK: - Private static properties
+
+	fileprivate static let isDebugBuild		: Bool = {
+		#if DEBUG
+		return true
+		#else
+		return false
+		#endif
+	}()
+
+	// MARK: - Private properties
+
+	fileprivate static var shared			: AppCenterSDK?
+	fileprivate var configuration			: Configuration?
+
+	// MARK: - Public properties
+
+	static var wasInitialized				: Bool {
+		return MSAppCenter.isConfigured()
+	}
+
+	// MARK: - Methods
+
+	/// This will setup the AppCenterSDK with the common base configuration. Crashes will be detected if the app is build with the release build type.
+	/// Distribution can be enabled using the configuration
+	///
+	/// - Parameters:
+	///   - configuration: Configuration object
+	static func setup(configuration: AppCenterSDK.Configuration = .default) {
+		guard (self.isDebugBuild == false || configuration.enableDebug == true) else {
+			return
+		}
+
+		let services = (configuration.isDistributionEnabled == true) ? [MSCrashes.self, MSDistribute.self] : [MSCrashes.self]
+
+		MSAppCenter.start(configuration.appSecret, withServices: services)
+		MSCrashes.setEnabled(configuration.enableCrashes)
+		MSCrashes.setDelegate(shared)
+	}
+
+	/// Returns True, if and only if the Service got started and is enabled.
+	var isDistributionEnabled: Bool {
+		return MSDistribute.isEnabled()
+	}
+
+	/// Will enable or disable the Distribution Feature of AppCenter
+	/// While disabling works always.
+	/// For enabling the Setup (aka. start) - Method should be called before.
+	///
+	/// Flow in the App, for Apps that want to dynamically change that State:
+	/// - Call the Setup Method for with `isDistributionEnabled` set to true.
+	/// - Disable Distribution using `enableDistribution(enabled: false)`
+	/// - Enable Distribution at a later time using the same method
+	///
+	/// - Parameter enabled: Enable or Disable Distribtion
+	func enableDistribution(enabled: Bool = true) {
+		MSDistribute.setEnabled(enabled)
+	}
+
+	/// This will create a `fatalError` to crash the app.
+	static func performTestCrash() {
+
+		MSCrashes.generateTestCrash()
+	}
+}
+
+extension AppCenterSDK {
+
+	struct Configuration {
+
+		fileprivate var enableDebug				: Bool
+		fileprivate var appSecret				: String
+		fileprivate var isDistributionEnabled	: Bool
+		fileprivate var enableCrashes			: Bool
+
+		/// Initializes a AppCenterSDK Configuration
+		///
+		///	- Parameters:
+		///		- appSecret: supply this manually if you dont want it in the info.plist
+		///		- enableDebug: Should start the Services even for Debug, Default is false
+		///		- isDistributionEnabled: Should start the Distribution Service, Default is false for Debug and Live apps, else true,
+		init(appSecret: String? = nil, enableDebug: Bool = false, isDistributionEnabled: Bool? = nil, enableCrashes: Bool = true) {
```

Comment:
We should have the option to disable the log attachment, like it is done on the Hockey version. 
-->`isLogUploadEnabled`
It can have a default value to `true`, like in the Hockey version